src/librustdoc/html/static/rustdoc.css

```diff
@@ -54,6 +54,21 @@
 		  box-sizing: border-box;
 }
 
+/* This part handles the "default" theme being used depending on the system one. */
+html {
+	content: "";
```

Comment:
Maybe use a custom css property instead: https://developer.mozilla.org/en-US/docs/Web/CSS/--*

Reply:
You mean CSS variable? Not supported on IE so we can't use it.

Reply:
IE doesn't support `prefers-color-scheme` anyway. IE would not set the custom property to any value, but the javascript has a fallback for that already.

Reply:
I'd prefer avoid potential failures if possible.

Reply:
Just to confirm, since string values for `content` is only ever allowed for `::before` and `::after`, this won't cause a situation where the entire page is replaced with whatever is set here? MDN seems to imply this, but i want to double-check that that's what's actually going on.

Reply:
The content is a hack that isn't displayed on this tag. You can only "see" it with js.

Reply:
What are the chances of this behavior changing in the future? I don't want to be trapped in a situation where the CSS spec starts allowing element replacement with strings, and a whole bunch of docs are rendered unusable because we started depending on this workaround.

Reply:
I'm not sure to follow you. There is nothing in the specs or anywhere changing this behavior for the moment and if there is any change, we'll have quite a lot of time before needing to update it. Also, what are you referring to when talking about "and a whole bunch of docs are rendered unusable because we started depending on this workaround."?

Reply:
If this behavior changes, i.e. some browser decides that setting `content` to a string on an element will cause the element to be replaced by that string, then any documentation already built by a rustdoc using this CSS will only display the string set here.

As long as the specs are on our side, and setting `content` to a string on an element does nothing, then i think i can be okay with this.